### ISSUE TITLE
Where are the APIs documented for passing a graph and getting back a layout ?

### ISSUE THREAD

**Issue by ymolists** [eclipse/elk#949]
Hi All. Thank you so much for writing and updating this library.

I have a server (written in go) that needs to layout a fairly large model. I am trying to see what is the best way to integrate with elk layout algorithms. At first i am thinking that i can get away by wrapping the code in java with a rest endpoint (in java as well) that receives the layout as  ELK's Xtext-based graph and converts the graph to json. As such i am trying to find java code examples that take a graph and return a json object so i can wrap it with a rest endpoint. Or any pointers you may give to achieve the same.

I first looked at elklive thinking that it was already doing that behind the scene. Unfortunately (mostly due to my limited js skills) i only see that it is embedding the elkjs directly and not calling (unless i am majorly mistaken) a rest endpoint to a server written in java. 

Are there java examples that show how to take a layout in elkt and output json or any data structure ? This will really make my integration and wrapping the code much more easier to overcome.

Best Regards

Best Regards

**Comment by soerendomroes** [comment]
The progress monitor has functionality to log a graph: https://github.com/eclipse/elk/blob/56270d73f92bbfebd42a1478f9057f2532a6bf67/plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/BasicProgressMonitor.java#L426

Is this what you are searching for?

**Comment by ymolists** [comment]
> Is this what you are searching for?

Hi @soerendomroes . Thank you for your answer.

This looks more like a debugging tool more than anything in my cursory look. Unless i am greatly mistaken.

I came across [this](https://github.com/eclipse/elk/issues/859#issuecomment-1216554131) comment of yours . which led me [ here ](https://github.com/eclipse/elk/blob/master/test/org.eclipse.elk.alg.test/src/org/eclipse/elk/alg/test/PlainJavaInitialization.java)

Basically that is what i am trying to do. Create a java main application that can call the elk layout directly and get a graph i can convert to json in return.

Before i go and chase how to do this based on the links above are there any examples that you know that could help ? I am hoping that maybe now there is an api/examples that use the layout using plain java. Even a command line java main example is fine.

Best Regards

**Comment by soerendomroes** [comment]
> Basically that is what i am trying to do. Create a java main application that can call the elk layout directly and get a graph i can convert to json in return.

Maybe the graph conversion tool [here](https://rtsys.informatik.uni-kiel.de/elklive/conversion.html) might be more the thing you are looking for:
https://github.com/kieler/elk-live/blob/e63def61a18600dd6f947b347a2635044fad8185/server/elkgraph-web/src/main/java/de/cau/cs/kieler/elkgraph/web/ElkGraphConversions.xtend#L176

**Comment by soerendomroes** [comment]
You could also have a look at the two graph conversion handlers for elkt and elkj [here](https://github.com/eclipse/elk/blob/master/plugins/org.eclipse.elk.graph.text.ui/src/org/eclipse/elk/graph/text/ui/ConvertGraphHandler.xtend) and [here](https://github.com/eclipse/elk/blob/master/plugins/org.eclipse.elk.graph.json.text.ui/src/org/eclipse/elk/graph/json/text/ui/ConvertGraphHandler.xtend).
